tests/setup.py

```diff
@@ -67,7 +80,7 @@ def _probe_ssh_connection(conn):
                 ready = True
 
         except SSHException as e:
-            if not str(e).endswith("Connection reset by peer"):
+            if not (str(e).endswith("Connection reset by peer") or str(e).endswith("Error reading SSH protocol banner")):
```

Comment:
Is the timeout 20 seconds? I would consider adding quite a bit more here. VMs with horrible performance is a quite common occurrence, so should have some leeway.

Reply:
I can add more, sure.

But just for you to get the full picture: there is already a 3 minutes timeout for the system to boot; this 20 seconds extra is only from the image reaching the login prompt to the ssh server being ready.